### PR TITLE
[CS-3999] Adds PIN migration to Change Pin UI

### PR DESCRIPTION
**Pull request by oidoug**
### Description

This PR adds new callback in useWalletManger for updating PIN for selected wallet.

- [x] Completes #(CS-3999)

### Checklist

- [x] Tested on a small device
- [x] Tested on iOS
- [x] Tested on Android

### Screenshots

https://user-images.githubusercontent.com/129619/176020287-e87207cd-e659-40e2-b0d7-cfb06c95bbba.mp4

### CODE REVIEW

File: src/hooks/useWalletManager.ts [hunk 1]
```diff
@@ -76,6 +79,18 @@ export default function useWalletManager() {
...
 
+  const updateWalletPIN = useCallback(async () => {
+    navigate(Routes.PIN_SCREEN, {
+      flow: PinFlow.new,
+      variant: 'light',
+      canGoBack: true,
+      dismissOnSuccess: true,
+      onSuccess: async (pin: string) => {
+        await migrateSecretsWithNewPin(selectedWallet, pin);
+      },
```
Comment (danibonilha) [inline src/hooks/useWalletManager.ts]: This is not the correct behavior, what we want here, is, before setting the pin, we need to decrypt the seed phrase, which uses the current pin, and then save it under the new PIN encryption. the `migrateSecretsWithNewPin`. should not be used anywhere outside the initial flow

**Review by danibonilha — COMMENTED** [review]
Apologies for not adding a description on the ticket, I've updated it, ping me if you have any doubts

File: src/helpers/walletLoadingStates.ts [hunk 2]
```diff
@@ -7,4 +7,5 @@ export default {
   RESTORING_WALLET: 'Restoring...',
   SWITCHING_ACCOUNT: 'Switching account...',
   MIGRATING_SECRETS: 'Migrating secrets...',
+  UPDATING_PIN: 'Updating PIN...',
```
Comment (danibonilha) [inline src/helpers/walletLoadingStates.ts]: This value is never used

Comment (oidoug) [reply src/helpers/walletLoadingStates.ts]: Right, the intention was to have a loading overlay but it's a quick process.

File: src/hooks/useWalletManager.ts (hunk 1 above)
Comment (oidoug) [reply src/hooks/useWalletManager.ts]: Right, I got carried away by the existing method working 😅 . 
I created a function in secure-storage to iterate wallets and resave seed, private keys and pin with new pin, please review the PR again.

**Comment by oidoug** [comment]
I just saw an issue, will update the PR soon.

File: src/components/settings-menu/SettingsSection.js [hunk 3]
```diff
@@ -165,7 +158,7 @@ export default function SettingsSection({
             <ListItem
```
Comment (danibonilha) [inline src/components/settings-menu/SettingsSection.js]: We need to move this ListItem, out of the  ` {__DEV__ && (` condition.

**Review by danibonilha — APPROVED** [review]
LGTM, just need to enable for all users, instead of just DEV

File: src/components/settings-menu/SettingsSection.js (hunk 3 above)
Comment (oidoug) [reply src/components/settings-menu/SettingsSection.js]: Ah yes, thanks.

I also removed the Seed Phrase Backup list item that has already been moved to a final implementation.

**Pull request merged**